docs/extend/modular-resources/upload/_index.md

```diff
@@ -231,3 +237,60 @@ You can also use the [Viam CLI](/manage/cli/) to update an existing custom modul
+
+1. Navigate to the **Actions** tab of the GitHub repository you are using for your module code.
+   If you have already created GitHub actions for this repository, click the **New workflow** button to create a new one.
+   If you have not yet created any GitHub actions, click the **Set up a workflow yourself** link.
+   See the [GitHub actions documentation](https://docs.github.com/en/actions/creating-actions) for more information.
+
+1. Paste the following action template YAML into the edit window.
+
+   ```yaml {class="line-numbers linkable-line-numbers"}
+   on:
+     push:
+     release:
+       types: [released]
+
+   jobs:
+     publish:
+       runs-on: ubuntu-latest
+       steps:
+       - uses: actions/checkout@v3
+       - name: build
+         run: echo "your build command goes here" && false # <-- replace this with the command that builds your module's tar.gz
+       - uses: viamrobotics/upload-module@main
+         # if: github.event_name == 'release' # <-- once the action is working, uncomment this so you only upload on release
+         with:
+           module-path: module.tar.gz
+           org-id: your-org-id-uuid # <-- replace with your org ID. not required for public modules
+           platform: linux/amd64 # <-- replace with your target architecture, or your module will not deploy
+           version: ${{ github.event_name == 'release' && github.ref_name || format('0.0.0-{0}.{1}', github.ref_name, github.run_number) }} # <-- see 'Versioning' section below for explanation
+           key-id: ${{ secrets.viam_key_id }}
+           key-value: ${{ secrets.viam_key_value }}
+   ```
+
+1. Edit the copied code to include the configuration specific to your module.
+   Each item marked with a `<--` comment requires that you edit the configuration values accordingly.
+
+   Set `run` to the command you use to build and package your module.
+   When ready to test the action, uncomment `if: github.event_name == 'release'` to enable the action to trigger a run when you [issue a release](https://docs.github.com/en/repositories/releasing-projects-on-github).
+
+   For guidance on configuring the other parameters, see the documentation for each:
+   - [`org-id`](/manage/cli/#using-the---org-id-and---public-namespace-arguments) - Not required if your module is public.
+   - [`platform`](/manage/cli/#using-the---platform-argument) - You can only upload one platform at a time.
+   - [`version`](/manage/cli/#using-the---version-argument)
+   - [`key-id` and `key-value`](/manage/cli/#create-an-organization-api-key) - You will set these values in the next step.
+
+1. Next, follow the instructions to [Create an organization API key](/manage/cli/#create-an-organization-api-key), which the action will use to authenticate its run.
+   Provide the `key-id` and `key-value` generated in the YAML configuration above.
+
+1. Push a commit to your module or [create a new release](https://docs.github.com/en/repositories/releasing-projects-on-github).
```

Comment:
'each commit' vs 'only github release' vs 'only tag release' workflows are subtly different

I didn't doc too deeply in the action's README.md because this is a general github thing, not a viam thing. your call whether to say anything about it here, but FYI

Reply:
Thank you, added a "your specific steps may differ" callout. 👍